c2corg_ui/views/document.py

```diff
@@ -413,20 +421,43 @@ def _get_api_cache_key_from_etag(self, etag):
     def _get_response(self, page_html):
         return get_response(self.request, page_html)
 
+    def _redirect(self, id, lang, slug=None):
+        location = ''
+        if slug is None:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view_id_lang', id=id, lang=lang)
+        else:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view', id=id, lang=lang, slug=slug)
+        raise HTTPMovedPermanently(location=location)
```

Comment:
> How did you check this? Our model does not prevent that redirects_to can not be changed anymore if it has been set once.

Because merging documents is a moderator-only feature, undoing a merge requires an hand-change in the DB in v5 and I'm pretty sure such a operation has never been requested (there are very few people that have ssh access to the server).

Of course documents might be merged by error and rollback would then be necessary. But this will happen very rarely (this is not a public feature). And the page ranking issue is much more important.

> The redirected page will also be indexed with 302. This graphic explains it quite well: https://seohackercdn-seohacker.netdna-ssl.com/wp-content/uploads/2011/10/301-and-302-Redirect-small2.png

Well, the bottom of the graphic explains that 
<img width="477" alt="capture d ecran 2016-08-31 a 17 18 14" src="https://cloud.githubusercontent.com/assets/1192331/18134725/aba444f6-6f9f-11e6-8358-3839317a7a93.png">

In order to actually transfer the indexing of the old page to the new one, a 301 is recommended.